Data for this notebook comes from https://collegescorecard.ed.gov/data/

In [23]:
# create dataframe for each raw data file to explore data

In [24]:
import pandas as pd
df_2013=pd.read_csv('rawdata/MERGED2013_PP.csv')
df_2013[:3]



UNITID    OPEID  opeid6                               INSTNM        CITY  \
0   100654   100200    1002             Alabama A & M University      Normal   
1   100663   105200    1052  University of Alabama at Birmingham  Birmingham   
2   100690  2503400   25034                   Amridge University  Montgomery   

  STABBR         ZIP                                       AccredAgency  \
0     AL       35762  Southern Association of Colleges and Schools C...   
1     AL  35294-0110  Southern Association of Colleges and Schools C...   
2     AL  36117-3553  Southern Association of Colleges and Schools C...   

                     INSTURL  \
0              www.aamu.edu/   
1                www.uab.edu   
2  www.amridgeuniversity.edu   

                                              NPCURL         ...          \
0     galileo.aamu.edu/netpricecalculator/npcalc.htm         ...           
1    www.collegeportraits.org/AL/UAB/estimator/agree         ...           
2  tcc.noellevitz.com/(S(miwoihs5stz5cpyifh4nczu0...         ...           

   PELL_RPY_3YR_RT_SUPP  NOPELL_RPY_3YR_RT_SUPP  FEMALE_RPY_3YR_RT_SUPP  \
0        0.443586698337           0.45251396648          0.484818805093   
1        0.732297852901          0.795454545455          0.760521885522   
2        0.600877192982                   0.775          0.610465116279   

   MALE_RPY_3YR_RT_SUPP  FIRSTGEN_RPY_3YR_RT_SUPP  \
0                 0.406            0.423580786026   
1        0.755344418052            0.751552795031   
2        0.686567164179            0.632653061224   

   NOTFIRSTGEN_RPY_3YR_RT_SUPP  C150_L4_POOLED_SUPP  C150_4_POOLED_SUPP  \
0               0.442452830189                  NaN    0.30871831265508   
1               0.765074770863                  NaN    0.50854980789381   
2                         0.65                  NaN   PrivacySuppressed   

   C200_L4_POOLED_SUPP  C200_4_POOLED_SUPP  
0                  NaN    0.35168556430446  
1                  NaN     0.4730923834537  
2                  NaN   PrivacySuppressed  

[3 rows x 1729 columns]

there are 1729 columns. probaby not all of them are helpful in deduplication, so we can drop many.

to figure out which ones to drop or keep, list them all:

In [25]:
list(df_2013.columns.values)

['\xef\xbb\xbfUNITID',
 'OPEID',
 'opeid6',
 'INSTNM',
 'CITY',
 'STABBR',
 'ZIP',
 'AccredAgency',
 'INSTURL',
 'NPCURL',
 'sch_deg',
 'HCM2',
 'main',
 'NUMBRANCH',
 'PREDDEG',
 'HIGHDEG',
 'CONTROL',
 'st_fips',
 'region',
 'LOCALE',
 'locale2',
 'LATITUDE',
 'LONGITUDE',
 'CCBASIC',
 'CCUGPROF',
 'CCSIZSET',
 'HBCU',
 'PBI',
 'ANNHI',
 'TRIBAL',
 'AANAPII',
 'HSI',
 'NANTI',
 'MENONLY',
 'WOMENONLY',
 'RELAFFIL',
 'ADM_RATE',
 'ADM_RATE_ALL',
 'SATVR25',
 'SATVR75',
 'SATMT25',
 'SATMT75',
 'SATWR25',
 'SATWR75',
 'SATVRMID',
 'SATMTMID',
 'SATWRMID',
 'ACTCM25',
 'ACTCM75',
 'ACTEN25',
 'ACTEN75',
 'ACTMT25',
 'ACTMT75',
 'ACTWR25',
 'ACTWR75',
 'ACTCMMID',
 'ACTENMID',
 'ACTMTMID',
 'ACTWRMID',
 'SAT_AVG',
 'SAT_AVG_ALL',
 'PCIP01',
 'PCIP03',
 'PCIP04',
 'PCIP05',
 'PCIP09',
 'PCIP10',
 'PCIP11',
 'PCIP12',
 'PCIP13',
 'PCIP14',
 'PCIP15',
 'PCIP16',
 'PCIP19',
 'PCIP22',
 'PCIP23',
 'PCIP24',
 'PCIP25',
 'PCIP26',
 'PCIP27',
 'PCIP29',
 'PCIP30',
 'PCIP31',
 'PCIP38',
 'PCIP39'

I think I'm interested in columns 0-25, so keep only those

In [26]:
df_2013 = df_2013.ix[:,0:25]
list(df_2013.columns.values)

['\xef\xbb\xbfUNITID',
 'OPEID',
 'opeid6',
 'INSTNM',
 'CITY',
 'STABBR',
 'ZIP',
 'AccredAgency',
 'INSTURL',
 'NPCURL',
 'sch_deg',
 'HCM2',
 'main',
 'NUMBRANCH',
 'PREDDEG',
 'HIGHDEG',
 'CONTROL',
 'st_fips',
 'region',
 'LOCALE',
 'locale2',
 'LATITUDE',
 'LONGITUDE',
 'CCBASIC',
 'CCUGPROF']

do the same with the remaining csv files and compare column headers

In [30]:
df_2012=pd.read_csv('rawdata/MERGED2012_PP.csv')
df_2012 = df_2012.ix[:,0:25]
df_2011=pd.read_csv('rawdata/MERGED2011_PP.csv')
df_2011 = df_2011.ix[:,0:25]
df_2010=pd.read_csv('rawdata/MERGED2010_PP.csv')
df_2010 = df_2010.ix[:,0:25]
df_2009=pd.read_csv('rawdata/MERGED2009_PP.csv')
df_2009 = df_2009.ix[:,0:25]
df_2008=pd.read_csv('rawdata/MERGED2008_PP.csv')
df_2008 = df_2008.ix[:,0:25]
df_2007=pd.read_csv('rawdata/MERGED2007_PP.csv')
df_2007 = df_2007.ix[:,0:25]
df_2006=pd.read_csv('rawdata/MERGED2006_PP.csv')
df_2006 = df_2006.ix[:,0:25]
df_2005=pd.read_csv('rawdata/MERGED2005_PP.csv')
df_2005 = df_2005.ix[:,0:25]
df_2004=pd.read_csv('rawdata/MERGED2004_PP.csv')
df_2004 = df_2004.ix[:,0:25]
df_2003=pd.read_csv('rawdata/MERGED2003_PP.csv')
df_2003 = df_2003.ix[:,0:25]
df_2002=pd.read_csv('rawdata/MERGED2002_PP.csv')
df_2002 = df_2002.ix[:,0:25]
df_2001=pd.read_csv('rawdata/MERGED2001_PP.csv')
df_2001 = df_2001.ix[:,0:25]
df_2000=pd.read_csv('rawdata/MERGED2000_PP.csv')
df_2000 = df_2000.ix[:,0:25]
df_1999=pd.read_csv('rawdata/MERGED1999_PP.csv')
df_1999 = df_1999.ix[:,0:25]
df_1998=pd.read_csv('rawdata/MERGED1998_PP.csv')
df_1998 = df_1998.ix[:,0:25]
df_1997=pd.read_csv('rawdata/MERGED1997_PP.csv')
df_1997 = df_1997.ix[:,0:25]
df_1996=pd.read_csv('rawdata/MERGED1996_PP.csv')
df_1996 = df_1996.ix[:,0:25]



In [28]:
common_cols = [col for col in set(df_2013.columns).intersection(df_2012.columns)]
print len(common_cols)
common_cols = [col for col in set(df_2012.columns).intersection(df_2011.columns)]
print len(common_cols)
common_cols = [col for col in set(df_2010.columns).intersection(df_2011.columns)]
print len(common_cols)
common_cols = [col for col in set(df_2010.columns).intersection(df_2009.columns)]
print len(common_cols)
common_cols = [col for col in set(df_2008.columns).intersection(df_2009.columns)]
print len(common_cols)
common_cols = [col for col in set(df_2008.columns).intersection(df_2007.columns)]
print len(common_cols)
common_cols = [col for col in set(df_2006.columns).intersection(df_2007.columns)]
print len(common_cols)
common_cols = [col for col in set(df_2006.columns).intersection(df_2005.columns)]
print len(common_cols)
common_cols = [col for col in set(df_2004.columns).intersection(df_2005.columns)]
print len(common_cols)
common_cols = [col for col in set(df_2004.columns).intersection(df_2003.columns)]
print len(common_cols)
common_cols = [col for col in set(df_2002.columns).intersection(df_2003.columns)]
print len(common_cols)
common_cols = [col for col in set(df_2002.columns).intersection(df_2001.columns)]
print len(common_cols)
common_cols = [col for col in set(df_2000.columns).intersection(df_2001.columns)]
print len(common_cols)
common_cols = [col for col in set(df_2000.columns).intersection(df_1999.columns)]
print len(common_cols)
common_cols = [col for col in set(df_1998.columns).intersection(df_1999.columns)]
print len(common_cols)
common_cols = [col for col in set(df_1998.columns).intersection(df_1997.columns)]
print len(common_cols)
common_cols = [col for col in set(df_1996.columns).intersection(df_1997.columns)]
print len(common_cols)

25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25
25


all columns are common, so we can merge the data into a single dataframe
but first, we also want to add a column for year to each df

In [32]:
df_2013['year']=2013
df_2012['year']=2012
df_2011['year']=2011
df_2010['year']=2010
df_2009['year']=2009
df_2008['year']=2008
df_2007['year']=2007
df_2006['year']=2006
df_2005['year']=2005
df_2004['year']=2004
df_2003['year']=2003
df_2002['year']=2002
df_2001['year']=2001
df_2000['year']=2000
df_1999['year']=1999
df_1998['year']=1998
df_1997['year']=1997
df_1996['year']=1996



now we can concatenate all dfs

In [33]:
frames = [df_2013, df_2012, df_2011, df_2010, df_2009, df_2008, df_2007, df_2006, df_2005, df_2004, df_2003, df_2002, df_2001, df_2000, df_1999, df_1998, df_1997, df_1996]

df_all = pd.concat(frames)

now we can add a unique identifier column

In [ ]:
df_all.iterrows
df_all['Id']=

now we can export the dataframe to csv with headers, and without indexes

In [34]:
df_all.to_csv('allyears.csv',index=False,header=True)